src/main/java/org/testobject/test/LogentriesResultReporter.java

```diff
@@ -0,0 +1,69 @@
+package org.testobject.test;
+
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LogentriesResultReporter extends TestWatcher {
+	private static final String LOGENTRIES_URL = "http://webhook.logentries.com/noformat/logs/";
+
+	private static final String JSON_TEST_FIELD = "test";
+	private static final String JSON_DURATION_FIELD = "duration";
+
+	private final Client client = ClientBuilder.newBuilder().build();
+
+	private final WebTarget webTarget;
+
+	private final Map<String, Long> testStartTimes = new HashMap<>();
+
+	public LogentriesResultReporter(String logentriesToken) {
+		this.webTarget = client.target(LOGENTRIES_URL).path(logentriesToken);
+	}
+
+	public LogentriesResultReporter() {
+		String logentriesToken = System.getenv("LOGENTRIES_TOKEN");
+		if (logentriesToken == null) {
+			throw new IllegalStateException("LOGENTRIES_TOKEN env was empthy. Please set the env variable");
+		}
+		this.webTarget = client.target(LOGENTRIES_URL).path(logentriesToken);
+	}
+
+	@Override
+	protected void starting(Description description) {
+		if (description.isEmpty()) {
+			return;
+		}
+		testStartTimes.put(description.getDisplayName(), System.currentTimeMillis());
+	}
+
+	@Override
+	protected void finished(Description description) {
+		if (description.isEmpty()) {
+			return;
+		}
+
+		Long startTime = testStartTimes.get(description.getDisplayName());
+		if (startTime != null) {
+			sendResult(description.getDisplayName(), startTime);
+		}
+		client.close();
+	}
+
+	private void sendResult(String displayName, long startTime) {
+		long testDuration = System.currentTimeMillis() - startTime;
+		Map<String, Object> json = new HashMap<>();
+		json.put(JSON_TEST_FIELD, displayName);
+		json.put(JSON_DURATION_FIELD, testDuration);
+		Response response = webTarget.request().post(Entity.json(json));
+		if (response.getStatus() != 204) {
```

Comment:
Does it make sense to throw an exception for any non 204 response code? What if Logentries changes to return 200?